packages/discord.js/src/client/websocket/WebSocketManager.js

```diff
@@ -22,13 +22,15 @@ const BeforeReadyWhitelist = [
   GatewayDispatchEvents.GuildMemberRemove,
 ];
 
-const UNRECOVERABLE_CLOSE_CODES = [
-  GatewayCloseCodes.AuthenticationFailed,
-  GatewayCloseCodes.InvalidShard,
-  GatewayCloseCodes.ShardingRequired,
-  GatewayCloseCodes.InvalidIntents,
-  GatewayCloseCodes.DisallowedIntents,
-];
+const unrecoverableErrorCodeMap = {
+  [GatewayCloseCodes.AuthenticationFailed]: ErrorCodes.TokenInvalid,
+  [GatewayCloseCodes.InvalidShard]: ErrorCodes.ShardingInvalid,
+  [GatewayCloseCodes.ShardingRequired]: ErrorCodes.ShardingRequired,
+  [GatewayCloseCodes.InvalidIntents]: ErrorCodes.InvalidIntents,
+  [GatewayCloseCodes.DisallowedIntents]: ErrorCodes.DisallowedIntents,
+};
+const UNRECOVERABLE_CLOSE_CODES = Object.keys(unrecoverableErrorCodeMap).map(k => Number(k));
```

Comment:
Instead of doing `UNRECOVERABLE_CLOSE_CODES.includes(code)`, can't we do `code in unrecoverableErrorCodeMap` or perhaps `Object.hasKey(unrecoverableErrorCodeMap, code)`?